yt/frontends/ramses/hilbert.py

```diff
@@ -60,51 +61,10 @@ def hilbert3d(
       The bit_length for the indexing.
     """
     X = np.atleast_2d(X)
-
-    x_bit_mask, y_bit_mask, z_bit_mask = (
-        np.zeros(bit_length, dtype=bool) for _ in range(3)
-    )
-    i_bit_mask = np.zeros(3 * bit_length, dtype=bool)
-
-    npoint = X.shape[0]
-    order = np.zeros(npoint)
-
-    # Convert positions to binary
-    for ip in range(npoint):
-        for i in range(bit_length):
-            mask = 0b01 << i
-            x_bit_mask[i] = X[ip, 0] & mask
-            y_bit_mask[i] = X[ip, 1] & mask
-            z_bit_mask[i] = X[ip, 2] & mask
-
-        for i in range(bit_length):
-            # Interleave bits
-            i_bit_mask[3 * i + 2] = x_bit_mask[i]
-            i_bit_mask[3 * i + 1] = y_bit_mask[i]
-            i_bit_mask[3 * i] = z_bit_mask[i]
-
-        # Build Hilbert ordering using state diagram
-        cstate = 0
-        for i in range(bit_length - 1, -1, -1):
-            sdigit = (
-                4 * i_bit_mask[3 * i + 2]
-                + 2 * i_bit_mask[3 * i + 1]
-                + 1 * i_bit_mask[3 * i]
-            )
-            nstate = _STATE_DIAGRAM[sdigit, 0, cstate]
-            hdigit = _STATE_DIAGRAM[sdigit, 1, cstate]
-
-            i_bit_mask[3 * i + 2] = hdigit & 0b100
-            i_bit_mask[3 * i + 1] = hdigit & 0b010
-            i_bit_mask[3 * i] = hdigit & 0b001
-
-            cstate = nstate
-
-        # Compute ordering
-        for i in range(3 * bit_length):
-            order[ip] = order[ip] + i_bit_mask[i] * 2**i
-
-    return order
+    # A note here: there is a freedom in the way hilbert indices are
+    # being computed (should it be xyz or yzx or zxy etc.)
+    # and the yt convention is not the same as the RAMSES one.
+    return get_hilbert_indices(bit_length, X[:, [1, 2, 0]])
```

Comment:
i'm not quite sure why only the windows test failed, but `get_hilbert_indices` is expecting an int array, 

https://github.com/yt-project/yt/blob/65ae2c4cfccaeab3dd8fbf751cf5eabceb8f1c37/yt/utilities/lib/geometry_utils.pyx#L267

Reply:
oh, saw that this PR is still a draft, so you probably noticed this already.

Reply:
Thanks for the notification, the type hints were actually erroneous in the version that's currently on main!